### ISSUE TITLE
Allow custom commit messages using `@@dolt_transaction_commit = 1`

### ISSUE THREAD

**Issue by Moosieus** [dolthub/dolt#7417]
Commits derived from transactions using [`@@dolt_transaction_commit = 1`](https://docs.dolthub.com/sql-reference/version-control/dolt-sysvars#dolt_transaction_commit) currently default `dolt_log.message` to `Transaction commit`. It'd be awesome to have a way to customize this.

Perhaps `DOLT_MESSAGE "Updating patient dob";` or some work-alike? I think `SAVEPOINT` in MySQL and Postgres may have the most kindred considerations in terms of functionality and syntax.

```
+----------------------------+
| message                    |
+----------------------------+
| Transaction commit         |
+----------------------------+
```

**Comment by timsehn** [comment]
We should make it so you can set it as a static string or pass the contents of the transaction being executed to a procedure and have that return a string to make the message. That might be two feature requests.

**Comment by fulghum** [comment]
Cool feature request. Thanks for opening this one with us @Moosieus. 

I like the idea of starting with a system variable (such as `@@dolt_transaction_commit_message`) during the transaction then on `commit` that message gets used for the Dolt commit, instead of the default message. The stored procedure sounds like a great follow up to allow even more customization. 

I'll go ahead and get started on this one for you and will keep you updated on progress.

**Comment by fulghum** [comment]
@Moosieus – we just released [Dolt 1.32.4](https://github.com/dolthub/dolt/releases/tag/v1.32.4) and it contains support for the new system variable `@@dolt_transaction_commit_message` that controls the commit message when `@@dolt_transaction_commit` is enabled. There's some [more info in the docs](https://docs.dolthub.com/sql-reference/version-control/dolt-sysvars#dolt_transaction_commit_message), too.

I'll go ahead and resolve this issue, but I still like the idea of a stored procedure that can generate the commit message, too. Feel free to send us a request that feature also, if you think it'd help out your app. 

Thanks for taking the time to share this feature request! 🙏  Let us know if there's anything else we can do to help you build with Dolt.

**Comment by bpf120** [comment]
@Moosieus , thanks for the feature request! We'd love to learn more about your Dolt use case too.  If you'd like to share, feel free to [email me](mailto:brianf@dolthub.com) or swing by our [Discord](https://discord.gg/gqr7K4VNKe).
